### ISSUE TITLE
BlockMeta not found for blockId xxxxx

### ISSUE THREAD

**Issue by qin7972** [Alluxio/alluxio#12804]
**Alluxio Version:**
2.1.1

**Describe the bug**
When a hive table is mounted and accessed by presto, it will appear more frequently in about 6 minutes, and then there will be no more.The exception is thrown.
```
com.facebook.presto.spi.PrestoException: Error reading from alluxio://zk@xxx:2181/hive/warehouse/dm.db
/dwd/dt=20210104/000134_0 at position 65732295
        at com.facebook.presto.hive.orc.HdfsOrcDataSource.readInternal(HdfsOrcDataSource.java:77)
        at com.facebook.presto.orc.AbstractOrcDataSource.readFully(AbstractOrcDataSource.java:105)
        at com.facebook.presto.orc.AbstractOrcDataSource.readFully(AbstractOrcDataSource.java:96)
        at com.facebook.presto.orc.cache.StorageOrcFileTailSource.getOrcFileTail(StorageOrcFileTailSource.java:59)
        at com.facebook.presto.orc.OrcReader.<init>(OrcReader.java:97)
        at com.facebook.presto.orc.OrcReader.<init>(OrcReader.java:75)
        at com.facebook.presto.hive.orc.OrcBatchPageSourceFactory.createOrcPageSource(OrcBatchPageSourceFactory.java:240)
        at com.facebook.presto.hive.orc.OrcBatchPageSourceFactory.createPageSource(OrcBatchPageSourceFactory.java:157)
        at com.facebook.presto.hive.HivePageSourceProvider.createHivePageSource(HivePageSourceProvider.java:309)
        at com.facebook.presto.hive.HivePageSourceProvider.createPageSource(HivePageSourceProvider.java:132)
        at com.facebook.presto.spi.connector.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:51)
        at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:58)
        at com.facebook.presto.operator.ScanFilterAndProjectOperator.getOutput(ScanFilterAndProjectOperator.java:227)
        at com.facebook.presto.operator.Driver.processInternal(Driver.java:379)
        at com.facebook.presto.operator.Driver.lambda$processFor$8(Driver.java:283)
        at com.facebook.presto.operator.Driver.tryWithLock(Driver.java:675)
        at com.facebook.presto.operator.Driver.processFor(Driver.java:276)
        at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:1077)
        at com.facebook.presto.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:162)
        at com.facebook.presto.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:545)
        at com.facebook.presto.$gen.Presto_0_232_46ebc4c__0_232____20201125_091504_1.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: alluxio.exception.status.InvalidArgumentException: BlockMeta not found for blockId 4849462656630784
        at alluxio.exception.status.AlluxioStatusException.from(AlluxioStatusException.java:127)
        at alluxio.exception.status.AlluxioStatusException.fromStatusRuntimeException(AlluxioStatusException.java:210)
        at alluxio.AbstractClient.retryRPCInternal(AbstractClient.java:385)
        at alluxio.AbstractClient.retryRPC(AbstractClient.java:345)
        at alluxio.client.block.RetryHandlingBlockMasterClient.getBlockInfo(RetryHandlingBlockMasterClient.java:118)
        at alluxio.client.block.AlluxioBlockStore.getInStream(AlluxioBlockStore.java:169)
        at alluxio.client.file.FileInStream.positionedReadInternal(FileInStream.java:255)
        at alluxio.client.file.FileInStream.positionedRead(FileInStream.java:235)
        at alluxio.hadoop.HdfsFileInputStream.read(HdfsFileInputStream.java:128)
        at alluxio.hadoop.HdfsFileInputStream.readFully(HdfsFileInputStream.java:145)
        at org.apache.hadoop.fs.FSDataInputStream.readFully(FSDataInputStream.java:107)
        at org.apache.hadoop.fs.FSDataInputStream.readFully(FSDataInputStream.java:107)
        at com.facebook.presto.cache.CachingInputStream.readFully(CachingInputStream.java:53)
        at com.facebook.presto.hive.orc.HdfsOrcDataSource.readInternal(HdfsOrcDataSource.java:64)
        ... 23 more
Caused by: alluxio.shaded.client.io.grpc.StatusRuntimeException: INVALID_ARGUMENT: BlockMeta not found for blockId 4849462656630784
        at alluxio.shaded.client.io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:233)
        at alluxio.shaded.client.io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:214)
        at alluxio.shaded.client.io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:139)
        at alluxio.grpc.BlockMasterClientServiceGrpc$BlockMasterClientServiceBlockingStub.getBlockInfo(BlockMasterClientServiceGrpc.java:557)
        at alluxio.client.block.RetryHandlingBlockMasterClient.lambda$getBlockInfo$3(RetryHandlingBlockMasterClient.java:120)
        at alluxio.AbstractClient.retryRPCInternal(AbstractClient.java:383)
        ... 34 more
```

**To Reproduce**
Not clear.

**Expected behavior**
No

**Urgency**
MEDIUM

**Additional context**
HeapBlockStore is used.

**Comment by calvinjia** [comment]
@qin7972 Thanks for reporting this issue. I had a few questions -
1. Does this issue no longer show up after a few minutes? 
2. Are there any other problems reading the data afterward?
3. Is the table being accessed for the first time?
4. Could you try to do `bin/alluxio fs ls -R <table path>` before running the query?

**Comment by qin7972** [comment]
@calvinjia 

1. Does this issue no longer show up after a few minutes?
    Yes, it does.
2. Are there any other problems reading the data afterward?
    No
3. Is the table being accessed for the first time?
   No. If data is updating in hive table, we will remount the table again, and then it will report the error occasionally.
4. Could you try to do bin/alluxio fs ls -R <table path> before running the query?
    The `bin/alluxio fs ls -R <table path>` not be executed when querying from presto through the alluxio client.

**Comment by calvinjia** [comment]
Thanks for the update.

How are you remounting the table? I suspect what is happening is there is already a set of data associated with those file paths, however after the data is updated, we will need to refresh the data blocks of those files. This might conflict with some other jobs which have a view of the old data blocks, and when they try to access those blocks they are not found (since we are refreshing the data).

**Comment by qin7972** [comment]
@calvinjia Hello, calvinjia.  We use a flag to switch whether it is available or not. When the data is updated, we set the flag to false, and presto directly accesses hive. Then we delete all the old table data, and remount it again. When remount is completed, we will set the flag to true, and presto accesses the alluxio. It will appear the error in few minutes. Not every time, occasionally.

**Comment by calvinjia** [comment]
When you say `we delete all the old table data, and remount it again`, what operations do you run on Alluxio?

**Comment by qin7972** [comment]
@calvinjia  If table A is updated, we will unmount all data path of table A, and remount the hive path of table A to the given alluxio path.

**Comment by jiacheliu3** [comment]
@qin7972 @calvinjia See https://github.com/Alluxio/alluxio/issues/11214, this has been fixed by higher version of Alluxio >=2.2.0.

**Comment by jja725** [comment]
Will close it for now, feel free to reopen it and contact us if this is still valid.
